test/e2e/framework/framework.go

```diff
@@ -473,6 +474,38 @@ func (f *Framework) AfterEach() {
 	}
 }
 
+// DeleteNamespace can be used to delete a namespace. Additionally it can be used to
+// dump namespace information so as it can be used as an alternative of framework
+// deleting the namespace towards the end.
+func (f *Framework) DeleteNamespace(name string) {
+	defer func() {
+		err := f.ClientSet.CoreV1().Namespaces().Delete(context.TODO(), name, metav1.DeleteOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			Logf("error deleting namespace %s: %v", name, err)
+			return
+		}
+		err = WaitForNamespacesDeleted(f.ClientSet, []string{name}, DefaultNamespaceDeletionTimeout)
+		if err != nil {
+			Logf("error deleting namespace %s: %v", name, err)
+			return
+		}
+		// remove deleted namespace from namespacesToDelete map
+		for i, ns := range f.namespacesToDelete {
```

Comment:
Hmm, I am wondering is it possible another test that has the same namespace getting created between the gap. And by remove that namespace from namespacesToDelete, we are leaking resources?

Reply:
same comment as below (https://github.com/kubernetes/kubernetes/pull/94318#discussion_r479613474) , ginko/k8s runs tests parallely in different processes not in different goroutines.